Sources/KituraWebSocket/WebSocketConnection.swift

```diff
@@ -244,6 +259,16 @@ extension WebSocketConnection: ChannelInboundHandler {
         }
     }
 
+    public func userInboundEventTriggered(context: ChannelHandlerContext, event: Any) {
+        guard event is IdleStateHandler.IdleStateEvent else { return }
+        if self.waitingForPong {
+            _ = context.channel.close(mode: .all)
+        } else {
+            self.sendMessage(with: .ping, data: context.channel.allocator.buffer(capacity: 2))
+            self.waitingForPong = true
+        }
+    }
+
     public func errorCaught(context: ChannelHandlerContext, error: Error) {
```

Comment:
here I would definitely recommend that on unknown error you either:
- forward (if this is _not_ the "application handler", ie. the last and the most high-level handler)
- log & close the `Channel` (in case this _is_ the last handler which I often call the "application handler")

Reply:
Thanks @weissi I've addressed this comment.